src/Simplex/Chat/Store.hs

```diff
@@ -1629,6 +1637,16 @@ mergeContactRecords db userId Contact {contactId = toContactId} Contact {contact
   deleteContactProfile_ db userId fromContactId
   DB.execute db "DELETE FROM contacts WHERE contact_id = ? AND user_id = ?" (fromContactId, userId)
   DB.execute db "DELETE FROM display_names WHERE local_display_name = ? AND user_id = ?" (localDisplayName, userId)
+  where
+    toFromContacts :: Contact -> Contact -> (Contact, Contact)
+    toFromContacts c1 c2
+      | directOrUsed c1 && not (directOrUsed c2) = (c1, c2)
+      | directOrUsed c2 && not (directOrUsed c1) = (c2, c1)
+      | otherwise = olderYoungerContacts c1 c2
+    olderYoungerContacts c1 c2
+      | ctCreatedAt c1 <= ctCreatedAt c2 = (c1, c2)
+      | otherwise = (c2, c1)
```

Comment:
```suggestion
      | not d1 && d2 = (c2, c1)
      | d1 == d2 && ctCreatedAt c1 > ctCreatedAt c2 = (c2, c1)
      | otherwise = (c1, c2)
    d1 = directOrUsed c1
    d2 = directOrUsed c2
```